lib/reporter.express.js

```diff
@@ -245,10 +246,13 @@ module.exports = function (reporter, definition) {
       return
     }
 
+    const initialApp = definition.options.app || express()

```

Comment:
i don't like that you create other express app here for all cases, it should only be done when subpath is set, i would move all the logic to `configureExpressApp` function and just create this initialApp there only when subpath is set, when it is not set it should continue with the same app that we had before. moving the logic to `configureExpressApp` will also make you not need to do `reporter.express.app = initialApp` bellow, you can do it right in `configureExpressApp`, that is important because we emit some events and we need to ensure that `reporter.express.app` is set once to avoid getting different express app in the events and when using the value directly from `reporter.express.app`